### PR TITLE
(maint) Add option to build dynamic libraries

### PR DESCRIPTION
**Pull request by MikaelSmith**
It's quite likely we'll want to build dynamic libraries of Leatherman
modules. Add that capability as an option to build instead of static
libraries, disabled by default.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
For fun, build and install dynamic libraries, download https://root.cern.ch/drupal/content/cling, and run
```
[cling]$ .I /usr/local/include
[cling]$ .L /usr/local/lib/libleatherman_curl.dylib
[cling]$ #include <leatherman/curl/client.hpp>
[cling]$ #include <iostream>
[cling]$ using namespace leatherman::curl;
[cling]$ using namespace std;
[cling]$ client http;
[cling]$ auto r = request("http://www.example.com");
[cling]$ auto resp = http.get(r);
[cling]$ cout << resp.body();
```

File: cmake/internal.cmake [hunk 1]
```diff
@@ -78,10 +78,17 @@ endmacro()
 # This macro cannot be invoked multiple times
 macro(add_leatherman_library)
     include_directories(${${include_var}})
-    add_library(${libname} STATIC ${ARGV})
+    if(LEATHERMAN_SHARED)
+        add_library(${libname} SHARED ${ARGV})
+        target_link_libraries(${libname} ${${deps_var}})
```
Comment (MikaelSmith) [inline cmake/internal.cmake]: Might be able to clear `${deps_var}` after doing this. I haven't thought it through.

Comment (branan) [inline cmake/internal.cmake]: I wouldn't. Not all platforms will implicitly pull in the dependencies of a shared object at link time, so we still need the deps for the FindLeatherman step (at least sometimes on old/weird unixes)

**Comment by MikaelSmith** [comment]
Updated.

**Comment by MikaelSmith** [comment]
After LTH-61, make which libnowide configurable based on the new flag.

**Comment by peterhuene** [comment]
Does this need to go in before or after PR #107?  Just checking before I merge this.

**Comment by MikaelSmith** [comment]
I can update #107 after it's merged.

**Pull request merged**